packages/transport-bluetooth/src/client/bluetooth-api.ts

```diff
@@ -0,0 +1,116 @@
+import {
+    AbstractApi,
+    AbstractApiConstructorParams,
+    DEVICE_TYPE,
+} from '@trezor/transport/src/api/abstract';
+import * as ERRORS from '@trezor/transport/src/errors';
+import { PathInternal } from '@trezor/transport/src/types';
+import { readMessageBuffer } from '@trezor/transport/src/utils/readMessageBuffer';
+
+import { TrezorBluetooth } from './trezor-bluetooth';
+import { BluetoothDevice } from './types';
+
+// implementation of @trezor/transport/src/api/AbstractApi
+
+type BluetoothApiParams = AbstractApiConstructorParams & {
+    url: string;
+};
+
+export class BluetoothApi extends AbstractApi {
+    chunkSize = 244;
+    api: TrezorBluetooth;
+    private readBuffer = readMessageBuffer();
+
+    constructor(options: BluetoothApiParams) {
+        super(options);
+
+        this.api = new TrezorBluetooth({ url: options.url, logger: options.logger });
+    }
+
+    private devicesToDescriptors(devices: BluetoothDevice[]) {
+        return devices
+            .filter(device => device.connected && device.paired)
+            .map(device => ({
+                path: device.id as PathInternal,
+                type: DEVICE_TYPE.TypeBluetooth,
+                id: device.id,
+            }));
+    }
+
+    async init() {
+        const { api } = this;
+        try {
+            await api.connect();
+        } catch (error) {
+            return this.error({ error: ERRORS.UNEXPECTED_ERROR, message: error.message });
+        }
+
+        const transportApiEvent = ({ devices }: { devices: BluetoothDevice[] }) => {
+            this.emit('transport-interface-change', this.devicesToDescriptors(devices));
+        };
+
+        api.on('device_connected', transportApiEvent);
+        api.on('device_disconnected', transportApiEvent); // TODO: readbuffer clear
+        api.on('device_read', ({ id, data }) => {
+            this.readBuffer.onMessage(id, Buffer.from(data));
+        });
+        api.on('adapter_state_changed', ({ state }) => {
+            if (state !== 'enabled') {
+                transportApiEvent({ devices: [] });
+            }
+        });
+        // TODO: discuss should it try to reconnect?
```

Comment:
probably yes 🤷 what problems could it introduce? 🤔

Reply:
more like what problem does it solve :)

if you kill `trezor-bluetooth` process (using task manager) then `BluetoothTransport` doesnt know it,  similar case to
```
// bridge
export const HTTP_ERROR = 'Network request failed' as const;
```

i ended up with adding another expected error const
```
export const API_DISCONNECTED = 'Api disconnected' as const;
```

Reply:
ok, maybe they may be unified both into API_DISCONNECTED. @marekrjpolak could be interested in this.